src/test.c

```diff
@@ -1,39 +1,53 @@
 #include "blake2b.h"
+#include "blake2b_kat.h"
 #include <stdio.h>
 #include <string.h>
 
-#define BUF_LENGTH 256
-
 void
-print_hex(const uint8_t* hash)
+print_hex(const uint8_t* hash, char* string, int len)
 {
   size_t i;
 
-  i=0;
-  printf("hash  :");
-  while (i++ < BLAKE2B_OUTBYTES) {
+  i = 0;
+  printf("%s: ", string);
+  while (i++ < len) {
     printf("%02x ", *hash++);
   }
+  printf("\n\n");
 }
 
 int
 main(int argc, char const* argv[])
 {
 
-  char buf[] = "abc";
-  char key[] = "";
+  uint8_t key[BLAKE2B_KEYBYTES];
+  uint8_t buf[BLAKE2_KAT_LENGTH];
   uint8_t hash[BLAKE2B_OUTBYTES];
-  size_t buflen = strlen(buf);
-  size_t keylen = strlen(key);
   size_t i;
 
-  for (i = 0; i <= buflen; ++i) {
-    blake2b(hash, BLAKE2B_OUTBYTES, buf, i, key, keylen);
+  /* keyed hashing not implemented yet */
+  for (i = 0; i < BLAKE2B_KEYBYTES; ++i)
+    key[i] = (uint8_t)i;
+
+  for (i = 0; i < BLAKE2_KAT_LENGTH; ++i)
+    buf[i] = (uint8_t)0;
+
+  for (i = 0; i < BLAKE2_KAT_LENGTH; ++i)
+    buf[i] = (uint8_t)i;
+
+  for (i = 0; i < BLAKE2_KAT_LENGTH; ++i) {
+    blake2b(hash, BLAKE2B_OUTBYTES, buf, i, key, BLAKE2B_KEYBYTES);
```

Comment:
It shouldn't really matter seeing as how there's no key. I'll make the change anyway.